### ISSUE TITLE
Drag-and-drop picture whose path contains "?" question/interrogation mark breaks path from the symbol and prevents embedding picture

### ISSUE THREAD

**Issue by hsandt** [zim-desktop-wiki/zim-desktop-wiki#2381]
**Describe the bug**
When drag and dropping a picture whose path contains a question mark `?`, the path is broken (stops after `?`, including `?` itself), breaking the image preview.

**To Reproduce**
1. Create a local folder `?` and put some image inside e.g. `picture.png`
2. Drag and drop the image inside Zim
3. Note how it adds a glitchy dot for small pictures, or an error symbol for big pictures
4. Open the Zim file in a text editor and see that the path was broken: instead of `{{~/Pictures/?/picture.png}}`, you'll see `{{~/Pictures/}}`

**Expected behavior**
Image preview should work with `{{~/Pictures/?/picture.png}}`

**Screenshots**
The glitchy dot at step 3:
![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/1067772/ef950363-2833-4cbe-a4a9-74abb6a321a9)

And for bigger dimensions:
![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/1067772/7d56662a-e45d-4b7c-bab3-88ced0f4cecc)

**Environment (please complete the following information):**
- Version: 0.75.1 (installed via Ubuntu PPA)
- OS: Linux Ubuntu 22.04
- Language: English, but URL is international

**Additional context**
The bug should occur if `?` is in the middle of the folder name too.
The bug doesn't occur if you drag and drop a non-picture file (it will just display the file path)
If you manually add the rest of the path inside the file, the picture will still not show, and next change on the page will crop the path again before the `?`

**Comment by introt** [comment]
`?` is not considered to be a valid character in file system filenames on most operating systems. While in theory it's not impossible to support, it would require a lot of extra logic that wouldn't work cross-platform. I'd recommend against using `?` in paths, as you're likely to run into problems if you ever need to use those files on other systems.

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/34b2d39a4aef83285e43840e12ffc47ec021e466/zim/notebook/page.py#L94-L110

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/34b2d39a4aef83285e43840e12ffc47ec021e466/zim/formats/wiki.py#L77-L78

**Comment by jaap-karssenberg** [comment]
@introt I agree it is not advisable to use a "?" in file paths. However, I'm investigating in what function the path is actually chopped. A quick test shows that the `os.path` functions allow a "?" so I don't see a good reason to chop the path (yet)

**Comment by hsandt** [comment]
I see, and yes it would be nice to find a particular reason to it, and better, trigger a clean error.

Indeed, while Zim will prevent a new page to be named with a `?` (changing text input background to red and disabling OK), dragging and dropping an image into Zim is always possible, and Linux can have `?` in paths. So this should be handled gracefully with an error message, if not supported.

By the way, *renaming* an page with `?` will not change text input BG to red and allow clicking OK, then you'd have a runtime error:

> This is zim 0.75.1
Platform: posix
Locale: en_US UTF-8
FS encoding: utf-8
Python: (3, 10, 6, 'final', 0)
PyGObject: (3, 42, 1)

> ======= Traceback =======
  File "/usr/lib/python3/dist-packages/zim/gui/widgets.py", line 3084, in do_response
    destroy = self.do_response_ok()
  File "/usr/lib/python3/dist-packages/zim/gui/uiactions.py", line 641, in do_response_ok
    newbasename = Path.makeValidPageName(name)
  File "/usr/lib/python3/dist-packages/zim/notebook/page.py", line 155, in makeValidPageName
    raise ValueError('Not a valid page name: %s (was: %s)' % (newname, name))
ValueError: Not a valid page name:  (was: ?)

I assume it's a bug?
